packages/core-app-api/src/apis/implementations/auth/pinniped/Pinniped.ts

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  AuthProviderInfo,
+  AuthRequestOptions,
+  ConfigApi,
+  PinnipedSupervisorApi,
+  pinnipedSupervisorApiRef,
+  DiscoveryApi,
+  OAuthRequestApi,
+} from '@backstage/core-plugin-api';
+import { OAuth2CreateOptions, OAuth2Session, PopupOptions } from '../oauth2';
+
+import { OAuth2Response } from '../oauth2/OAuth2';
+import {
+  DefaultAuthConnector,
+  RefreshingAuthSessionManager,
+  SessionManager,
+} from '../../../../lib';
+
+const DEFAULT_PROVIDER = {
+  id: 'pinniped',
+  title: 'Pinniped',
+  icon: () => null,
+};
+
+/**
+ *
+ *
+ * @public
+ */
+export default class Pinniped implements PinnipedSupervisorApi {
+  private audiences: { [aud: string]: SessionManager<OAuth2Session> };
+  private configApi: ConfigApi | undefined;
+  private environment: string;
+  private provider: AuthProviderInfo;
+  private oauthRequestApi: OAuthRequestApi;
+  private discoveryApi: DiscoveryApi;
+  private defaultScopes: string[];
+  private popupOptions: PopupOptions | undefined;
+
+  static create(
+    options: OAuth2CreateOptions,
+  ): typeof pinnipedSupervisorApiRef.T {
+    return new Pinniped(options);
+  }
+
+  private constructor(options: OAuth2CreateOptions) {
+    const {
+      configApi,
+      environment = 'development',
+      provider = DEFAULT_PROVIDER,
+      oauthRequestApi,
+      discoveryApi,
+      defaultScopes = [
+        'openid',
+        'offline_access',
+        'pinniped:request-audience',
+        'username',
+      ],
+      popupOptions,
+    } = options;
+
+    this.configApi = configApi;
+    this.environment = environment;
+    this.provider = provider;
+    this.oauthRequestApi = oauthRequestApi;
+    this.discoveryApi = discoveryApi;
+    this.defaultScopes = defaultScopes;
+    this.popupOptions = popupOptions;
+
+    this.audiences = {};
+  }
+
+  async getSupervisorIdToken(
+    _options?: AuthRequestOptions | undefined,
+  ): Promise<string> {
+    throw new Error('Method not implemented.');
+  }
+
+  async getClusterScopedIdToken(
+    audience: string,
+    options?: AuthRequestOptions | undefined,
+  ): Promise<string> {
+    const aud = await Pinniped.getAudience(audience);
```

Comment:
I don't think we need the `getAudience` helper at all; looks like maybe a copy-paste from the microsoft API?
```suggestion
    const aud = audience;
```